packages/vite/src/vite.ts

```diff
@@ -27,6 +28,23 @@ export async function bundle (nuxt: Nuxt) {
   const useAsyncEntry = nuxt.options.experimental.asyncEntry ||
     (nuxt.options.vite.devBundler === 'vite-node' && nuxt.options.dev)
   const entry = await resolvePath(resolve(nuxt.options.appDir, useAsyncEntry ? 'entry.async' : 'entry'))
+
+  let allowDirs = [
+    nuxt.options.appDir,
+    nuxt.options.workspaceDir,
+    ...nuxt.options._layers.map(l => l.config.rootDir),
+    ...nuxt.apps.default?.components.map(c => dirname(c.filePath)),
+    ...nuxt.apps.default?.plugins.map(p => dirname(p.src)),
+    ...nuxt.apps.default?.middleware.map(m => dirname(m.path)),
+    ...Object.values(nuxt.apps.default?.layouts || {}).map(l => dirname(l.file)),
+    dirname(nuxt.apps.default.rootComponent!),
+    dirname(nuxt.apps.default.errorComponent!)
+  ].filter(d => d && existsSync(d))
```

Comment:
I guess this list makes some duplicates (unnecessary nested paths `/foo` covers `/foo/bar`) but i guess should be fine for performance /cc @antfu ? (or a sort might help at least)

Reply:
This PR does handle nested paths - see:

https://github.com/nuxt/nuxt/blob/49bfa2d37984473727ad573b50ffe079558c6457/packages/vite/src/vite.ts#L44-L46

Deduplicating is certainly worth it though.